app/assets/stylesheets/overlay.scss

```diff
@@ -0,0 +1,234 @@
+$global-radius: 4px;
+
+
+.image-wrapper {
+  width: 100%;
+  height: 100%;
+  border: 1px solid rgba(0,0,0,0.04);
+  overflow: hidden;
+  position: relative;
+  text-align: center;
+  border-radius: $global-radius;
+}
+
+.image-overlay-content {
+  width: 100%;
+  height: 100%;
+  position: absolute;
+  overflow: hidden;
+  top: 0;
+  left: 0
+}
+
+// modifier classes
+
+.overlay-fade-in {
+
+  p {
+    letter-spacing: 0.15em;
+    color: #f4f4f4;
+    font-size: 28px;
+    opacity: 0;
+    transition: all 0.2s linear;
+  }
+
+  img {
+    transition: all 0.2s linear;
+  }
+
+  .image-overlay-content {
+    opacity: 0;
+    background-color: rgba(0,0,0, 0.4);
+    transition: all 0.4s ease-in-out;
+  }
+
+  h2 {
+    color: #f2f2f2;
+    font-size: 1.8rem;
+    margin-top: 35%;
+    opacity: 0;
+    transition: all 0.2s ease-in-out;
+    background: rgba(0,0,0,0.7);
+  }
+
+  .button {
+    display: inline-block;
+    text-decoration: none;
+    padding: 7px 14px;
+    background: #FFF;
+    color: #222;
+    text-transform: uppercase;
+    box-shadow: 0 0 1px #000;
+    position: relative;
+    border: 1px solid #999;
+    opacity: 0;
+    transition: all 0.2s ease-in-out;
+
+    &:hover {
+      box-shadow: 0 0 5px #000;
+    }
+  }
+
+  &:hover img {
+    transform: scale(1.2);
+  }
+
+  &:hover .image-overlay-content {
+    opacity: 1;
+  }
+
+  &:hover h2, p, .button {
+    opacity: 1;
+  }
+
+  &:hover p {
+    transition-delay: 0.1s;
+  }
+
+  &:hover .button {
+    transition-delay: 0.2s;
+  }
+}
+
+
+
+
+
+
+
+
+
+.overlay-slide-in-left {
+
+  img {
+    transition: all 0.3s ease-in-out;
+  }
+
+  .image-overlay-content {
+    background-image: url(http://www.awwwards.com/awards/images/2015/04/pattern.jpg);
+    background-color: rgba(0,0,0,0.3);
+    transform: translateX(-110%);
+    opacity: 1;
+    transition: all 0.4s ease-in-out;
+    box-shadow: 0 0 7px #ccc;
+  }
+
+  h2{
+    font-size: 1.8rem;
+    background: rgba(255, 255, 255, 0.7);
+    margin-top: 40%;
+    color: #000;
+    box-shadow: 0px 1px 1px rgba(0,0,0,0.1);
```

Comment:
Commas in function arguments should be followed by a single space<br>`0px` should be written without units as `0`